homeassistant/components/niko_home_control/light.py

```diff
@@ -73,7 +80,32 @@ def turn_off(self, **kwargs):
         """Instruct the light to turn off."""
         self._light.turn_off()
 
-    def update(self):
-        """Fetch new state data for this light."""
-        self._light.update()
-        self._state = self._light.is_on
+    async def async_update(self):
+        """Get the latest data from NikoHomeControl API."""
+        await self._data.async_update()
+        self._state = self._data.getState(self._light.id)
+
+
+class NikoHomeControlData:
+    """The class for handling data retrieval."""
+
+    def __init__(self, nhc):
+        """Set up Niko Home Control Data object."""
+        self._nhc = nhc
+        self.available = True
+        self.data = {}
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self):
+        """Get the latest data from the NikoHomeControl API."""
+        try:
+            self.data = await self.hass.async_add_executor_job(
+                self._nhc.list_actions_raw())
```

Comment:
Sorry my typo in previous comment 
```suggestion
                self._nhc.list_actions_raw)
```